app/services/company.js

```diff
@@ -15,28 +16,38 @@ class CompanyService extends CrudService {
       error.throwValidationError('Company already exists.');
     }
 
-    if (company.fieldOfActivityId !== undefined) {
-      const fieldExists = await fieldOfActivityService.get(company.fieldOfActivityId);
+    const { business, ...newCompany } = company;
+    if (business !== undefined) {
+      const fieldExists = await fieldOfActivityRepository.getByPropsNonParanoid({ name: business });
```

Comment:
This is not the best name for this variable. Here I consider it a boolean value, but a few lines below I see that it is in fac an object. I would call it "fieldOfActivity".